DetoxSync/DetoxSync/Spies/NSTimer+DTXSpy.m

```diff
@@ -116,11 +119,20 @@ void __detox_sync___NSCFTimer_invalidate(NSTimer* timer)
 	//	NSLog(@"🤦‍♂️ invalidate: %@", timer);
 	
 	id<DTXTimerProxy> trampoline = [DTXTimerSyncResource existingTimerProxyWithTimer:timer];
-	[trampoline untrack];
-	
 	__orig___NSCFTimer_invalidate(timer);
+
+	[trampoline untrack];
 }
 
+- (void)dealloc
+{
+    id<DTXTimerProxy> trampoline = [DTXTimerSyncResource existingTimerProxyWithTimer:self];
+
+    if(trampoline) {
+        NSLog(@"🤦‍♂️ dealloc, but trampoline was still active: %@", self);
+        [trampoline untrack];
+    }
+}
```

Comment:
I think adding this makes it safe to remove `_DTXCleanTimersAndReturnCount` (since we've then got a nicer way of calling untrack if the timer is `dealloc`ed)?

Reply:
Make sense.